clang/test/SemaTemplate/deduction-guide.cpp

```diff
@@ -261,6 +261,13 @@ AG ag = {1};
 // CHECK:   | `-BuiltinType {{.*}} 'int'
 // CHECK:   `-ParmVarDecl {{.*}} 'int'
 
+template <typename X = int>
+using BG = G<int>;
+BG bg(1.0);
+// CHECK-LABEL: Dumping <deduction guide for BG>
+// CHECK:      FunctionTemplateDecl {{.*}} implicit <deduction guide for BG>
+// CHECK: |-CXXDeductionGuideDecl {{.*}} 'auto (int) -> G<int>' aggregate
```

Comment:
Nit: Since this isn't matching the structure of the AST anyway, it would probably be a good idea to avoid the possibility of unrelated test churn.

```suggestion
// CHECK: CXXDeductionGuideDecl {{.*}} 'auto (int) -> G<int>' aggregate
```

Reply:
The line is a CHECK, not a CHECK-NEXT, so it could appear any number of lines afterwards.
So this means it could be a child of something else that appears after it.

The CHECK matcher matches any number of characters before the first character specified,
So it would match something like `| | |-CXXDeductionGuideDecl {{.*}} 'auto (int) -> G<int>' aggregate`
Ie, it could be deeply nested into something else.

Instead, I think it just demonstrates:
* It is a child of something.
* It's neither the first nor the last child of that thing.

Reply:
> The line is a CHECK, not a CHECK-NEXT, so it could appear any number of lines afterwards.

Yeah, ideally it should be `CHECK-NEXT`, but we can't use it because of the ambiguity in the content of the above line (there are two other deduction guides emitting the same content), so I end up using the `CHECK: |-...`, it at least demonstrates that it is a child of something.

Reply:
But that it is a child of something is trivial, because all declarations are children of something, except a translation unit declaration.

But the fact that this line test that this thing is neither the first nor the last child of that thing, means that it is quite likely to break with unrelated test churn.